tests/system/python/test_e2e_kafka.py

```diff
@@ -0,0 +1,207 @@
+
+        # Assertions on Kafka topic, consumer and subscription
+        # GET topics
+        self._verify_kafka_topic(conn, topic)
+
+        # GET topic by name
+        self._verify_kafka_topic_by_name(conn, topic)
+
+        # Create a consumer for JSON data, starting at the beginning of the topic's
+        # The consumer group is called "CONSUMER_GROUP" and the instance is "CONSUMER_INSTANCE".
+        self._verify_kafka_consumer_group_and_instance(conn)
+
+        # Subscribe the consumer to a topic
+        self._verify_consumer_subscription_to_topic(conn, topic)
+
+        # Then consume some data from a topic using the base URL in the first response.
+        self._verify_consume_data_from_topic(conn)
+
+        # Finally, close the consumer with a DELETE to make it leave the group and clean up  its resources.
+        self._close_consumer(conn)
+
+    def _verify_kafka_topic(self, conn, topic):
+        conn.request("GET", '/topics')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        val = json.loads(r)
+        assert topic in val
+
+    def _verify_kafka_topic_by_name(self, conn, topic):
+        conn.request("GET", '/topics/{}'.format(topic))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        val = json.loads(r)
+        assert topic == val["name"]
+
+    def _verify_kafka_consumer_group_and_instance(self, conn):
+        data = {"name": CONSUMER_INSTANCE, "format": "json", "auto.offset.reset": "earliest"}
+        conn.request("POST", '/consumers/{}'.format(CONSUMER_GROUP), json.dumps(data), headers=HEADER)
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        val = json.loads(r)
+        assert CONSUMER_INSTANCE == val["instance_id"]
+
+    def _verify_consumer_subscription_to_topic(self, conn, topic):
+        data = {"topics": [topic]}
+        conn.request("POST", '/consumers/{}/instances/{}/subscription'.format(CONSUMER_GROUP, CONSUMER_INSTANCE),
+                     json.dumps(data), headers=HEADER)
+        r = conn.getresponse()
+        # No content response
+        assert 204 == r.status
+
+    def _verify_consume_data_from_topic(self, conn):
+        # FIXME: This is failing http.client.ResponseNotReady: Request-sent
+        # Does not work with wait time
+        conn.request("GET", '/consumers/{}/instances/{}/records'.format(CONSUMER_GROUP, CONSUMER_INSTANCE))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        val = json.loads(r)
+        # TODO assert {'sensor': SENSOR_VALUE} in val
+
+    def _close_consumer(self, conn):
+        # FIXME: This should execute after consumer group/instance creation if in between fails on assertion
+        header = {'Content-Type': 'application/vnd.kafka.v2+json', 'Accept': 'application/vnd.kafka.json.v2+json'}
```

Comment:
delete this and use `HEADER` but most importnat is to have it as a teardown (always delete) and may be remove assert